srcdeps-maven-local-repository/src/main/java/org/srcdeps/mvn/localrepo/SrcdepsLocalRepositoryManager.java

```diff
@@ -64,84 +61,15 @@
-        public ConfigurationHolder(final MavenSession session) {
-            super();
-            MavenProject topLevelProject = session.getTopLevelProject();
-            Path basePath = topLevelProject.getBasedir().toPath();
-            Path srcdepsYamlPath = basePath.resolve(relativeMvnSrcdepsYaml);
-            if (Files.exists(srcdepsYamlPath)) {
-                this.configurationLocation = srcdepsYamlPath;
-                log.debug("SrcdepsLocalRepositoryManager using configuration {}", configurationLocation);
-            } else {
-                throw new RuntimeException(
-                        String.format("Could not locate srcdeps configuration at [%s]", srcdepsYamlPath));
-            }
-
-            String encoding = session.getSystemProperties().getProperty("project.build.sourceEncoding");
-            if (encoding == null) {
-                encoding = session.getUserProperties().getProperty("project.build.sourceEncoding", "utf-8");
-            }
-
-            Charset cs = Charset.forName(encoding);
-            try (Reader r = Files.newBufferedReader(configurationLocation, cs)) {
-                this.configuration = new YamlConfigurationIo().read(r);
-            } catch (IOException | ConfigurationException e) {
-                throw new RuntimeException(e);
-            }
-
-        }
-
-        /**
-         * Finds the first {@link ScmRepository} associated with the given {@code artifact}. The association is given by
-         * the exact string match between the groupId of the {@code artifact} and one of the
-         * {@link ScmRepository#getSelectors() selectors} of {@link ScmRepository}
-         *
-         * @param artifact
-         * @return
-         */
-        public ScmRepository findScmRepo(Artifact artifact) {
-            final String groupId = artifact.getGroupId();
-            for (ScmRepository scmRepository : getConfiguration().getRepositories()) {
-                if (scmRepository.getSelectors().contains(groupId)) {
-                    return scmRepository;
-                }
-            }
-            throw new IllegalStateException(String
-                    .format("No srcdeps SCM repository configured in .mvn/srcdeps.yaml for groupId [%s]", groupId));
-        }
-
-        /**
-         * @return the {@link Configuration} loaded from {@link #configurationLocation}
-         */
-        public Configuration getConfiguration() {
-            return configuration;
-        }
-
-        /**
-         * Returns {@code ".mvn/srcdeps.yaml"} resolved against the top level project the current Maven request.
-         *
-         * @return
-         */
-        public Path getConfigurationLocation() {
-            return configurationLocation;
-        }
-
-    }
+    /** See the bin/mvn or bin/mvn.cmd script of your maven distro, where maven.multiModuleProjectDirectory is set */
+    private static final String maven_multiModuleProjectDirectory_PROPERTY = "maven.multiModuleProjectDirectory";
```

Comment:
As you may think, I tried to transform `"maven.multiModuleProjectDirectory"` (without too much thinking) to a valid constant name. You are right, there are at least two other conventions in place in this project. OK, I'll use one of the existing ones.